[dtensor] update some comments

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107608
* #107606

This update some comments from the follow up of https://github.com/pytorch/pytorch/pull/107305

wanchaol committed:
[dtensor] update some comments

This update some comments from the follow up of https://github.com/pytorch/pytorch/pull/107305

[ghstack-poisoned]

fduwjj commented on torch/distributed/_tensor/debug/__init__.py:
```diff
@@ -4,6 +4,8 @@
 def get_sharding_prop_cache_info():
     """
     Get the cache info for the sharding propagation cache, used for debugging purpose only.
+    This would return a named tuple showing hits, misses, maxsize and currsize of the sharding
```
nit: maybe `cursize`?

fduwjj reviewed (approved):
LGTM

wanchaol committed:
Update on "[dtensor] update some comments"


This update some comments from the follow up of https://github.com/pytorch/pytorch/pull/107305

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
